docs/src/modules/components/DemoSandbox.js

```diff
@@ -161,41 +162,36 @@ const jss = create({
  * Isolates the demo component as best as possible. Additional props are spread
  * to an `iframe` if `iframe={true}`.
  */
-function DemoSandboxed(props) {
+function DemoSandbox(props) {
   const router = useRouter();
-  const asPathWithoutLang = router.asPath.replace(/^\/[a-zA-Z]{2}\//, '/');
-  const { component: Component, iframe, name, onResetDemoClick, ...other } = props;
-  const Sandbox = iframe ? DemoFrame : React.Fragment;
+  const { canonicalAs } = pathnameToLanguage(router.asPath);
+  const { children: childrenProp, iframe = false, name, onResetDemoClick, ...other } = props;
+  const Sandbox = iframe ? DemoIframe : React.Fragment;
   const sandboxProps = iframe ? { name, ...other } : {};
 
   const t = useTranslate();
 
+  // `childrenProp` needs to be a child of `Sandbox` since the iframe implementation rely on `cloneElement`.
+  const children = <Sandbox {...sandboxProps}>{childrenProp}</Sandbox>;
+
   return (
     <DemoErrorBoundary name={name} onResetDemoClick={onResetDemoClick} t={t}>
```

Comment:
Actually there is ErrorBoundary in `react-runner`, I left it as it is for minimal change